AsyncDisplayKit/ASTextNode.mm

```diff
@@ -394,26 +400,23 @@ - (void)setAttributedText:(NSAttributedString *)attributedText
 
   // Tell the display node superclasses that the cached layout is incorrect now
   [self invalidateCalculatedLayout];
-
   [self setNeedsDisplay];
-  
-  
-  // Accessiblity
-  self.accessibilityLabel = _attributedText.string;
-  self.isAccessibilityElement = (_attributedText.length != 0); // We're an accessibility element by default if there is a string.
 }
 
 #pragma mark - Text Layout
 
 - (void)setExclusionPaths:(NSArray *)exclusionPaths
 {
-  std::lock_guard<std::recursive_mutex> l(_textLock);
+  {
+    std::lock_guard<std::recursive_mutex> l(_textLock);
   
-  if (ASObjectIsEqual(exclusionPaths, _exclusionPaths)) {
-    return;
+    if (ASObjectIsEqual(exclusionPaths, _exclusionPaths)) {
+      return;
+    }
+  
+    _exclusionPaths = [exclusionPaths copy];
   }
   
-  _exclusionPaths = [exclusionPaths copy];
```

Comment:
Same implication here regarding renderer and display invalidation, these operations may need to occur together as if they were a "transaction".